team-01/src/main/java/ubc/cosc322/MonteCarloPlayer.java

```diff
@@ -33,40 +33,42 @@
  * 2. Fullstack Academy, https://www.youtube.com/watch?v=Fbs4lnGLS8M
  */
 public class MonteCarloPlayer extends LocalPlayer {
-	private final long MAX_RUNTIME = 10000;
+	private final long MAX_RUNTIME = 25000;
 
 	// The constant used for UCB function. Same one chosen in the John Levine video.
 	private final double EXPLORATION_FACTOR = Math.sqrt(2);
 
 	private TreeNode root;
 
 	public MonteCarloPlayer() {
-		super("jaden_mc", "montecarlo");
+		super("group1-mc", "montecarlo");
 	}
 
 	@Override
 	protected void onMoveReceived() {
 		AmazonsLocalBoard rootState = board.copy();
-		rootState.localPlayer = board.getOpponent();
 		root = new TreeNode(rootState);
 
 		long startTime = System.currentTimeMillis();
 		long endTime = startTime + MAX_RUNTIME;
-
 		int count = 0;
+		int winner;
 		while (System.currentTimeMillis() < endTime) {
 			TreeNode current = getMaxLeaf(root);
-			if (current.getVisits() == 0) {
-				int winner = playthrough(current);
-				backpropagate(current, winner);
-			} else {
-				TreeNode child = current.expand();
-				int winner = playthrough(child);
-				backpropagate(child, winner);
+			TreeNode child = current.expand();
+
+			if (child == null) {
+				//unaccounted for terminal state here so we need to send back the results
+				backpropagate(current, current.state.getOpponent());
+				continue;
 			}
+
+			winner = playthrough(child);
+			backpropagate(child, winner);
+
 			count++;
```

Comment:
- This seems to have been the cause of our MCTS only moving one agent at a time, we were only looking at, we would pick one node and continue to explore it rather than exploring all nodes